[BottomAppBar] Tint leading, trailing bar items.

romoore opened the pull request:
The BottomAppBar should support different tint colors for the leading
and trailing items.

**Extreme Example**
<img width="371" alt="screen shot 2018-09-08 at 11 30 49 pm" src="https://user-images.githubusercontent.com/1753199/45260822-44407300-b3bf-11e8-85e0-a962ff969d97.png">


Part of #3928

Commit:
[NavigationBar] Tint leading, trailing button bars.

To prepare for tinting the leading and trailing button bar items in the
Bottom App Bar, the MDCNavigationBar should support an API that allows
separate tint colors for the leading and trailing button bar items.

Part of #3928

Commit:
Formatting nits

Commit:
[BottomAppBar] Tint leading, trailing bar items.

The BottomAppBar should support different tint colors for the leading
and trailing items.

Part of #3928

Commit:
Remove duplicate properties

randallli reviewed (dismissed):
lets add docs for tintColor to point at these two extra API for bar item tint colors so clients will know where to find them.

Commit:
Document the new behavior on `tintColor`

romoore commented:
`tintColor` documented in 8649232b811937256bfa21fed07efe2754fb39b6

Commit:
Restore tintColor assignment

Commit:
Suppress `tintColor` nullability warning.

Bazel builds are failing because redeclaring the property resulted in
some nullability checks failing. Since we're not expecting to
reimplement `tintColor` and only redeclare it for documenting it, the
warning should be suppressed.

Commit:
Remove redeclared `tintColor`.

I couldn't suppress the analyzer warning in clang, so I'm removing the
redelaration.
